framework/tests/value/MyTree.java

```diff
@@ -21,9 +21,16 @@ void uses() {
 
         MyTree<@UnknownVal String> myTree1 = newTree("hello").put("bye");
         // :: error: (assignment.type.incompatible)
+        MyTree<@StringVal("hello") String> myTree1b = newTree("hello").put("bye");
+
+        // Note: This is a false positive: the type newTree("hello").put("hello") is able to be
```

Comment:
How about `Note: This is a false positive: the type of newTree("hello").put("hello") should be inferred as MyTree<@StringVal("hello") String> and the assignment should therefore pass.`

Reply:
Updated. Thanks for the review!